elastic/kibana
# [Fleet] Fix preconfigured remote ES outputs with secrets

## Summary

Closes https://github.com/elastic/kibana/issues/172546

### Testing

1. Add the following to your Kibana config file and start Kibana:
```yml
xpack.fleet.outputs:
  - id: my-remote-es-output-with-a-secret
    name: preconfigured remote elasticsearch output with a secret
    type: remote_elasticsearch
    hosts: ['http://localhost:9997']
    secrets:
      service_token: secretRemoteESServiceToken
```
2. Head to Fleet UI and check that the output was correctly created.
3. Go to the Dev Tools Console and query Fleet secrets: `GET .fleet-secrets/_search`. Check that the service token secret was created.
4. Edit the output configuration, e.g. the value of the service token, and wait for Kibana to restart.
5. Check that the value(s) have been correctly updated.
